zerver/tests/test_narrow.py

```diff
@@ -413,6 +414,33 @@ def get_query_ids(self):
 
         return query_ids
 
+    def test_tag_highlighting(self):
+        # type: () -> None
+        """
+        Test if tags are highlighted properly
+        """
+
+        def run_highlight_match_tag_pos(string, match):
+            # type: (Text, Text) -> Tuple[int, int]
+            return highlight_match_tag_pos(string, 0, (string.find(match), len(match)), '<', '>')
+        def run_highlight_string_text_offsets(string, match):
+            # type: (Text, Text) -> Text
+            return highlight_string_text_offsets(string, [(string.find(match), len(match))])
+
+        self.assertEqual(run_highlight_match_tag_pos('text text <tag> text', 'tag'), (10, 5))
+        self.assertEqual(run_highlight_match_tag_pos('text <tag attr> text', 'attr'), (5, 10))
+        self.assertEqual(run_highlight_match_tag_pos('</tag>', 'tag'), (0, 6))
+        self.assertEqual(run_highlight_match_tag_pos('1<tag>', 'tag'), (1, 5))
+        self.assertEqual(run_highlight_match_tag_pos('<tag>text</tag>', 'text'), (5, 4))
+        self.assertEqual(run_highlight_string_text_offsets('<tag>text</tag>', 'text'), '<tag><span class="highlight">text</span></tag>')
+        self.assertEqual(run_highlight_string_text_offsets('stuff <tag attr="text"></tag> stuff', 'text'), 'stuff <span class="highlight"><tag attr="text"></span></tag> stuff')
+        self.assertEqual(run_highlight_string_text_offsets('<tag attr="text"></tag>', 'text'), '<span class="highlight"><tag attr="text"></span></tag>')
+        # Note: above 2 might not be really correct HTML, but detecting opening/closing tags is going to be too complicated for what it's worth
```

Comment:
I think we will probably want to support this eventually, but yeah, it's probably not essential (not a super likely search query in some sense).  I think it's good to have these tests to record known-broken cases.